pkg/controlplane/controller/clusterauthenticationtrust/cluster_authentication_trust_controller.go

```diff
@@ -149,8 +149,11 @@ func (c *Controller) syncConfigMap() error {
 	} else if err != nil {
 		return err
 	}
-	// keep the original to diff against later before updating
-	authConfigMap := originalAuthConfigMap.DeepCopy()
+
+	existingData, err := extractAPIServerOwnedFields(originalAuthConfigMap)
```

Comment:
Hmm... I'm really iffy on using field manager for this... is the field manager name stable and reliable? Shouldn't we just tolerate / preserve all unknown keys as-is to be forward-compatible?

Reply:
The writes on the CM this controller performs now explicitly set the manager name.

I wasn't sure we want to make the controller own the CM completely or whether we're ok with any unknown fields. Either way it's possible to add fields to the CM with enough privileges, with the managed fields they only have to be a bit more explicit about it.

If it's fine to keep any unknown fields, I'll remove the managed fields condition.